test/test_uop_graph.py

```diff
@@ -46,5 +46,145 @@ def test_const_cast(self):
     self.assertEqual(out.uop, UOps.CONST)
     self.assertEqual(out.arg, 0)
 
+  def test_loop_scope_load_local(self):
```

Comment:
Could you make this test more end-to-end?
Ideally, we'd want AST + List of optimizations that create this uop graph instead of handcoding it

Reply:
yeah bet, should i add them to #4220 somehow as well? master doesn't have the scheduler changes or linearizer changes to generate the patterns i want to test